src/cargo/ops/cargo_new.rs

```diff
@@ -772,12 +773,11 @@ fn mk(config: &Config, opts: &MkOptions<'_>) -> CargoResult<()> {
     if let Some(registry) = opts.registry {
         let mut array = toml_edit::Array::default();
         array.push(registry);
-        array.decor_mut().set_suffix("\n\n# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html");
         manifest["package"]["publish"] = toml_edit::value(array);
-    } else {
-        manifest["package"]["edition"].as_value_mut().expect("edition is a string value").decor_mut().set_suffix("\n\n# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html");
     }
-    manifest["dependencies"] = toml_edit::Item::Table(toml_edit::Table::new());
+    let mut dep_table = toml_edit::Table::default();
+    dep_table.decor_mut().set_prefix("\n# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html\n\n");
+    manifest["dependencies"] = toml_edit::Item::Table(dep_table);
 
```

Comment:
This looks like a refactor of the previous commit and unrelated to this commit.  Could you isolate the changes so we can more easily see what is related to the intent of the current commit?

Reply:
Fixed.